### PR TITLE
Add try_push_uninit and test

### PR DESCRIPTION
**Pull request by phughk**
Add try_push_uninit , which tries to add an additional element and provides a raw pointer to the already allocated memory.

This is useful if the backing structure is larger than the stack size.

Included is a test that demonstrates the need for this functionality.

### CODE REVIEW

File: src/arrayvec.rs [hunk 1]
```diff
@@ -206,6 +206,22 @@ impl<T, const CAP: usize> ArrayVec<T, CAP> {
         ArrayVecImpl::try_push(self, element)
     }
 
+    /// Push a new uninitialised value to the end of the vector and return
+    /// a mutable reference to it.
+    ///
+    /// This is useful if the backed value is large and won't fit on stack.
+    ///
+    /// Capacity error carries the data - since we don't have data then we need a different error type.
```
Comment (bluss) [inline src/arrayvec.rs:214]: please use `CapacityError<()>`

**Comment by bluss** [comment]
Something like this is needed, but unsure about the right API

**Comment by tbu-** [comment]
Perhaps this could take the same shape as [`Vec::spare_capacity_mut`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.spare_capacity_mut)?

**Comment by bluss** [comment]
@tbu- that sounds good!

**Comment by bluss** [comment]
@tbu-  was quick to add a suggestion in #279 which is a pretty nice interface - and more versatile in some ways - I hope that addresses the need here, what do you think?

**Comment by phughk** [comment]
@tbu- @bluss thank you! Yeah I'm not obsessed about the PR, I added this to a fork and used in my own project and wanted to contribute back. If you have a better API then very happy with the turnaround. Thank you as well for the diligent review; I learnt from it 😊

**Pull request closed without merging**